webapp/tests/Unit/Controller/Team/ProblemControllerTest.php

```diff
@@ -183,4 +186,43 @@ public function testInteractiveSamples(): void
         $response = $this->client->getResponse();
         self::assertEquals(404, $response->getStatusCode());
     }
+
+    /**
+     * Test that the problems page does not show page, statement, sample data before contest start.
+     */
+    public function testAccessProblemBeforeContestStarts(): void
+    {
+        $this->loadFixtures([DemoAboutToStartContestFixture::class]);
+
+        /** @var EntityManagerInterface $em */
+        $em = self::getContainer()->get(EntityManagerInterface::class);
+        $problemCount = $em->createQueryBuilder()
+            ->select('COUNT(cp.problem)')
+            ->from(ContestProblem::class, 'cp')
+            ->join('cp.contest', 'c')
+            ->where('c.shortname = :shortname')
+            ->setParameter('shortname', 'demo')
+            ->getQuery()
+            ->getSingleScalarResult();
+
+        $this->withChangedConfiguration('public_access_before_contest', true, function () use ($problemCount) {
+            $this->client->request('GET', '/public/problems');
+            static::assertSelectorTextContains('.nav-item .nav-link.disabled', 'Problems');
+            static::assertSelectorTextContains('.alert.alert-secondary', 'No problem texts available at this point.');
+            
+            for ($i = 1; $i <= $problemCount; $i++) {
```

Comment:
You can't hardcode the `$i` here as I think we use the global problemId and not the index in the contest, so when there are multiple problems in the database we might pick the wrong one.

Reply:
Changed to use contest_id to fetch pid from the problem.